tests/documentation/command-reference/doc_command_reference_init_test.go

```diff
@@ -0,0 +1,220 @@
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		helper.Chdir(commonVar.Context)
+		Expect(helper.VerifyFileExists(".odo/env/env.yaml")).To(BeFalse())
+	})
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+	Context("Interactive Mode", func() {
+		It("Empty directory", func() {
+			args := []string{"odo", "init"}
+			out, err := helper.RunInteractive(args, []string{"ODO_LOG_LEVEL=0"}, func(ctx helper.InteractiveContext) {
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "Java")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, "my-java-maven-app")
+
+				helper.ExpectString(ctx, "Your new component 'my-java-maven-app' is ready in the current directory")
+			})
+			Expect(err).To(BeNil())
+			got := fmt.Sprintf(outputStringFormat, " init", helper.StripSpinner(out))
+			got = helper.ReplaceTimeInString(got, "[1s]")
+			want := helper.GetMDXContent(filepath.Join(commonPath, "interactive_mode_empty_directory_output.mdx"))
+			want = helper.ReplaceTimeInString(want, "[1s]")
+			diff := cmp.Diff(want, got)
+			Expect(diff).To(BeEmpty())
+		})
+
+		When("the directory is not empty", func() {
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+			})
+
+			It("Directory with sources", func() {
+				args := []string{"odo", "init"}
+				out, err := helper.RunInteractive(args, []string{"ODO_LOG_LEVEL=0"}, func(ctx helper.InteractiveContext) {
+					helper.ExpectString(ctx, "Is this correct?")
+					helper.SendLine(ctx, "")
+
+					helper.ExpectString(ctx, "✓  Downloading devfile \"nodejs:2.1.1\" from registry \"DefaultDevfileRegistry\"")
+
+					helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+					helper.SendLine(ctx, "")
+
+					helper.ExpectString(ctx, "Enter component name")
+					helper.SendLine(ctx, "")
+
+					helper.ExpectString(ctx, "Changes will be directly reflected on the cluster.")
+				})
+				Expect(err).To(BeNil())
+				got := helper.StripAnsi(out)
+				got = fmt.Sprintf(outputStringFormat, "init", helper.StripSpinner(got))
+				got = strings.Replace(got, "? Is this correct? (Y/n) ", "", 1)
+				got = strings.Replace(got, "? Is this correct? Yes", "? Is this correct? Yes\n✓  Downloading devfile \"nodejs:2.1.1\" from registry \"DefaultDevfileRegistry\" [3s]", 1)
+				got = strings.Replace(got, "\n? Select container for which you want to change configuration?  [Use arrows to move, type to filter]\nruntime\n> NONE - configuration is correct\n", "", 1)
+				got = strings.Replace(got, "? Enter component name: (node-echo) ", "", 1)
```

Comment:
Could you add comments to explain the logic of these replace? I'm pretty sure they will be difficult to maintain

Reply:
Interactive mode is very unreliable in obtaining the same output every time. I am looking for a better way to do this.

Reply:
For the first iteration, I think you can ignore interactive commands, and focus on non-interactive commands

Reply:
Since I already have the tests, I am going to try and fix the failing ones. I'll mark them as skip if they still fail.